static-lang-word-lists/build.rs

```diff
@@ -0,0 +1,149 @@
+    io::{Cursor, Write},
+    path::{Path, PathBuf},
+    sync::Mutex,
+    thread,
+};
+
+use brotli::enc::{
+    backward_references::BrotliEncoderMode, BrotliEncoderParams,
+};
+use heck::ToShoutySnakeCase;
+
+// FIXME: change branch to `main` before merging
+const BASE_URL: &str = "https://raw.githubusercontent.com/googlefonts/fontheight/refs/heads/word-list-crate/static-lang-word-lists/data";
+
+// Provides WORD_LISTS: &[(&str, &str)] for word list name and relative paths
+// See egg.py for how this code is generated
+include!("chicken.rs");
+
+fn main() {
+    println!("cargo::rerun-if-changed=chicken.rs");
+    println!("cargo::rerun-if-changed=build.rs");
+    println!("cargo::rerun-if-env-changed=STATIC_LANG_WORD_LISTS_LOCAL");
+
+    if option_env!("STATIC_LANG_WORD_LISTS_LOCAL").is_some() {
+        println!("cargo::warning=building from local files");
+    }
+
+    let word_list_path = out_dir_path("word_list_codegen.rs");
+    let word_list_file = Mutex::new(open_path(&word_list_path));
+    let map_path = out_dir_path("map_codegen.rs");
+    let mut map_file = open_path(&map_path);
+
+    writeln!(
+        &mut map_file,
+        "pub static LOOKUP_TABLE: ::phf::Map<&'static str, &'static \
+         ::std::sync::LazyLock<::fontheight_core::WordList>> = \
+         ::phf::phf_map! {{"
+    )
+    .unwrap_or_else(|err| panic!("failed to write to map_codeden.rs: {err}"));
+    let map_file = Mutex::new(map_file);
+
+    thread::scope(|s| {
+        // Bind references to names so they can be copied to each spawned thread
+        let codegen_file = &word_list_file;
+        let map_file = &map_file;
+        WORD_LISTS.iter().copied().for_each(|(name, path)| {
+            s.spawn(move || {
+                let ident = name.to_shouty_snake_case();
+                let bytes =
+                    if option_env!("STATIC_LANG_WORD_LISTS_LOCAL").is_none() {
+                        download_validate(path)
+                    } else {
+                        let repo_path = Path::new("data").join(path);
+                        fs::read(&repo_path).unwrap_or_else(|err| {
+                            panic!(
+                                "failed to read local word list file {}: {err}",
+                                repo_path.display()
+                            );
+                        })
+                    };
+                let br_path = compress(&bytes, path);
+
+                let mut codegen_file = codegen_file.lock().unwrap();
+                writeln!(
+                    &mut codegen_file,
+                    r#"
+                    wordlist! {{
+                        ident: {ident},
+                        name: {name},
+                        bytes: include_bytes!(r"{}"),
+                    }}"#,
+                    br_path.display(),
+                )
+                .unwrap_or_else(|err| {
+                    panic!("failed to write to word_list_codegen.rs: {err}");
+                });
```

Comment:
Will this end up writing the contents of the file in a different order on each run, depending on which thread finishes first? If yes, could it be an issue in that the builds won't be "reproducible"? (e.g. if crates.io does some hashing of the binaries to check integrity or something like that?) I'm thinking about this sort of issues: https://github.com/fonttools/fonttools/issues/3120